src/main/java/net/kaczmarzyk/spring/data/jpa/utils/DateTimeUtils.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2014-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.utils;
+
+import java.sql.Timestamp;
+import java.time.*;
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+import java.util.Date;
+
+@SuppressWarnings("unchecked")
+public abstract class DateTimeUtils {
+
+	private static final String UTC_ZONE_ID = "UTC";
+
+	public static <T> T startOfDay(Object dateObject) {
+		if (dateObject instanceof Calendar) {
```

Comment:
used `instanceof` instead of `isAssignableFrom` to include subtypes and different implementations of classes like e.g `Calendar` -> implementations: `Gregorian` amd `Japanese`.

Reply:
hmm Calendar.class.isAssignableFrom(dateObject)  should handle it as well or...?